server/modules/authentication/discord/authentication.js

```diff
@@ -19,9 +32,17 @@ module.exports = {
         passReqToCallback: true
       }, async (req, accessToken, refreshToken, profile, cb) => {
         try {
-          if (conf.guildId && !_.some(profile.guilds, { id: conf.guildId })) {
+          if (conf.roles) {
+            const memberRoles = await discord.getGuildMember(accessToken, conf.guildId)
```

Comment:
There are two `await discord.getGuildMember` calls right now - let's remove the redundant one 😅

```suggestion
```